doc/reference/javascript_cheatsheet.rst

```diff
@@ -276,4 +276,237 @@ using the view in the kanban arch (a specific example is the helpdesk dashboard)
     Note: you can change the way the view interprets the arch structure.  However,
     from the server point of view, this is still a view of the same base type,
     subjected to the same rules (rng validation, for example).  So, your views still
-    need to have a valid arch field.
\ No newline at end of file
+    need to have a valid arch field.
+
+Promises and asynchroneous code
+===============================
+For a very good and complete introduction to promises, please read this excellent article https://github.com/getify/You-Dont-Know-JS/blob/master/async%20%26%20performance/ch3.md
+
+Creating a new Promises
+-----------------------
+- turn a constant into a promise
+    There is 2 static functions on Promise that create a resolved or rejected promise based on a constant:
+
+    .. code-block:: javascript
+
+        var p = Promise.resolve({blabla: '1'}); // creates a resolved promise
+        p.then(function (result) {
+                console.log(result); // --> {blabla: '1'};
+        });
+
+
+        var p2 = Promise.reject({error: 'error message'}); // creates a rejected promise
+        p2.catch(function (reason) {
+                console.log(reason); // --> {error: 'error message');
+        });
+
+
+    .. note:: Note that even if the promises are created already resolved or rejected, the `then` or `catch` handlers will still be called asynchronously.
+
+
+- based on an already asynchronous code
+    Suppose that in a function you must do a rpc, and when it is completed set the result on this.
+    The `this._rpc` is a function that returns a `Promise`.
+
+    .. code-block:: javascript
+
+        function callRpc() {
+            var self = this;
+            return this._rpc(...).then(function(result) {
+                self.myValueFromRpc = result;
+            });
+        }
+
+- for callback based function
+    Suppose that you were using a function `this.close` that takes as parameter a callback that is called when the closing is finished.
+    Now suppose that you are doing that in a method that must send a promise that is resolved when the closing is finished.
+
+    .. code-block:: javascript
+        :linenos:
+
+        function waitForClose() {
+            var self = this;
+            return new Promise (function(resolve, reject) {
+                self.close(resolve);
+            });
+        }
+
+    * line 2: we save the `this` into a variable so that in an inner function, we can access the scope of our component
```

Comment:
no sure it's worth talking about that, isn't it quite common js knowledge?

Reply:
anyway if we want to mention it, i'd do it on the first example